vars/buildPlugin.groovy

```diff
@@ -12,7 +12,13 @@ def call(Map params = [:])
 		durabilityHint('PERFORMANCE_OPTIMIZED'),
 		pipelineTriggers(env.BRANCH_NAME == 'master' ? [cron('@weekly')] : []),
 		[$class: 'RebuildSettings', autoRebuild: false, rebuildDisabled: true],
-		[$class: 'ThrottleJobProperty', categories: [], limitOneJobWithMatchingParams: false, maxConcurrentPerNode: 0, maxConcurrentTotal: 1, paramsToUseForLimit: '', throttleEnabled: true, throttleOption: 'category']
+		[$class: 'ThrottleJobProperty', categories: [], limitOneJobWithMatchingParams: false, maxConcurrentPerNode: 0, maxConcurrentTotal: 1, paramsToUseForLimit: '', throttleEnabled: true, throttleOption: 'category'],
+		parameters([
+			string(defaultValue: '1', description: 'debian package revision tag', name: 'TAGREV', trim: true),
+			choice(choices: ['all', 'cosmic', 'bionic', 'xenial'], description: 'Ubuntu version to build for', name: 'dists'),
```

Comment:
Can you try making this a multi select, then the all special handling could be removed.

Reply:
I tried, but couldn't get it to work. The pipeline generator doesn't seem to work for multi-select

Reply:
I noticed the same, can you try the following.

Add this before `properties` and then add `extendedChoice` instead of this line. See https://wiki.jenkins.io/display/JENKINS/Extended+Choice+Parameter+plugin?focusedCommentId=127107242#comment-127107242
```
import com.cwctravel.hudson.plugins.extended_choice_parameter.ExtendedChoiceParameterDefinition;

UBUNTU_VERSIONS = ["xenial", "bionic", "cosmic"]

def extendedChoice = new ExtendedChoiceParameterDefinition(
	"dists" /* String name */,
	ExtendedChoiceParameterDefinition.PARAMETER_TYPE_MULTI_SELECT /* String type */,
	UBUNTU_VERSIONS.join(",") /* String value */,
	null /* String projectName */,
	null /* String propertyFile */, 
	null /* String groovyScript */,
	null /* String groovyScriptFile */, 
	null /* String bindings */, 
	null /* String groovyClasspath */, 
	null /* String propertyKey */, 
	UBUNTU_VERSIONS.join(",") /* String defaultValue */, 
	null /* String defaultPropertyFile */,
	null /* String defaultGroovyScript */, 
	null /* String defaultGroovyScriptFile */, 
	null /* String defaultBindings */, 
	null /* String defaultGroovyClasspath */,
	null /* String defaultPropertyKey */, 
	null /* String descriptionPropertyValue */, 
	null /* String descriptionPropertyFile */, 
	null /* String descriptionGroovyScript */,
	null /* String descriptionGroovyScriptFile */, 
	null /* String descriptionBindings */, 
	null /* String descriptionGroovyClasspath */, 
	null /* String descriptionPropertyKey */,
	null /* String javascriptFile */,
	null /* String javascript */,
	false /* boolean saveJSONParameterToFile*/,
	false /* boolean quoteValue */,
	UBUNTU_VERSIONS.size(), /* int visibleItemCount */,
	"Ubuntu version to build for" /* String description */,
	null /* String multiSelectDelimiter */
)
```